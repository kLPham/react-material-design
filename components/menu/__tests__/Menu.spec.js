import React from 'react';
import { shallow } from 'enzyme';
import Menu from '../index';
import MenuItem from '../menuItem';

describe('<Menu />', () => {
    it('should render', () => {
        const label = 'Clickie';
        const menu = shallow(
            <Menu label={label}>
                <MenuItem value="One" />
            </Menu>,
    );
        expect(menu).toMatchSnapshot();
    });
    it('should render disabled', () => {
        const label = 'Clickie';
        const disabled = true;
        const menu = shallow(
            <Menu label={label} disabled={disabled}>
                <MenuItem value="One" />
                <MenuItem value="Two" />
            </Menu>,
    );
        expect(menu).toMatchSnapshot();
    });
});
