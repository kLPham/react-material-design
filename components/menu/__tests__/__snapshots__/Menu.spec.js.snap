// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render 1`] = `
<div>
  <Button
    className="mdc-menu-anchor"
    label="Clickie"
    onClick={[Function]}
  />
  <div
    className="mdc-simple-menu"
    tabIndex="-1"
  >
    <ul
      aria-hidden="true"
      className="mdc-simple-menu__items mdc-list"
      role="menu"
    >
      <MenuItem
        value="One"
      />
    </ul>
  </div>
</div>
`;

exports[`<Menu /> should render disabled 1`] = `
<div>
  <Button
    className="mdc-menu-anchor"
    disabled={true}
    label="Clickie"
    onClick={[Function]}
  />
  <div
    className="mdc-simple-menu"
    tabIndex="-1"
  >
    <ul
      aria-hidden="true"
      className="mdc-simple-menu__items mdc-list"
      role="menu"
    >
      <MenuItem
        value="One"
      />
      <MenuItem
        value="Two"
      />
    </ul>
  </div>
</div>
`;
